src/plugins/management/public/management_sections_service.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { ReactElement } from 'react';
+import { ManagementSection, RegisterManagementSectionArgs } from './utils';
+import { managementSections } from './components/management_sections';
+
+import { ManagementSectionId, SectionsServiceSetup, SectionsServiceStart } from './types';
+
+export class ManagementSectionsService {
+  private sections: Map<ManagementSectionId | string, ManagementSection> = new Map();
```

Comment:
@cjcenizal  Thank you for your question. I forgot to highlight that change. From my point of view it's a regression of #65796. Unfortunately we cannot use only `ManagementSectionId` for ids. 

Let me explain you one case: 
`ManagementSectionsService` returns `register` method in runtime setup contract
It means that I should be able to register custom Management Sections for my custom plugins. This functionality worked fine until #65796 and for user it looked like: 
![image](https://user-images.githubusercontent.com/20072247/83120531-db8cb300-a0d9-11ea-8d97-397d04eb3da4.png)

But now `register` method accept only  `ManagementSectionId` for ids. As a result we have a type error: 
![image](https://user-images.githubusercontent.com/20072247/83121063-856c3f80-a0da-11ea-8568-634c49b2c56f.png)

Reply:
@alexwizp Have we guaranteed to plugin developers that we support custom Management sections? My original thoughts on this change were that it was acceptable to introduce this change if the feature wasn't supported, and it was also preferable to disallow custom Management sections because it would encourage a proliferation of custom sections with one or two plugins in each one.

Reply:
@cjcenizal Not sure if I am the best person to answer this question. 
I can only say that if we do not support this functionality, the  register method  should be removed from the public runtime contract. 

But it seems to me that if I made a couple of my personal plugins for my personal needs, why can not I create a personal management section for them....

Reply:
@cjcenizal 

>Have we guaranteed to plugin developers that we support custom Management sections?

I don't think so.

I wouldn't mind getting rid of the ability to create new management sections but are we certain that ALL potential plugins would be covered by the existing categories? Or do we need some sort of catch all? Likely only used by 3rd party developers.